client/signup/steps/site-topic/index.jsx

```diff
@@ -158,14 +160,15 @@ class SiteTopicStep extends Component {
 const mapDispatchToProps = ( dispatch, ownProps ) => ( {
 	submitSiteTopic: siteTopic => {
 		const { flowName, goToNextStep } = ownProps;
+		const siteTopicSlug = toSiteTopicSlug( siteTopic );
 
 		dispatch(
 			recordTracksEvent( 'calypso_signup_actions_submit_site_topic', {
-				value: siteTopic,
+				value: siteTopicSlug,
 			} )
 		);
 
-		dispatch( submitSiteTopic( siteTopic ) );
+		dispatch( submitSiteTopic( siteTopicSlug ) );
```

Comment:
If we press the Back button to come back to the site topic step, then we will see the english value since `submitSiteTopic()` is setting `siteTopicSlug` instead of the translated value. I think we should set the untranslated `siteTopic` here, and then in _step-actions.js_ we can send the english value to the _sites_new_ endpoint by doing `toSiteTopicSlug()`.